### DISCUSSION TITLE
Proper delay-based or rollback netcode.

### DISCUSSION THREAD

**Discussion by Kikasuru** [ikemen-engine/Ikemen-GO#840] [Feature Request] [answered]
Hello! Is there a way you can update the netcode to a more conventional and reliable one other than the wait-each-frame-for-an-update code we have currently? That'd be greatly appreciated. 

Info on netcode types [here](http://wiki.shoryuken.com/Netcode)
GGPO's C++ library [here](https://github.com/pond3r/ggpo)
GGPO is a popular implementation of rollback netcode for fighting games.

~~I would also like to ask if you could fix the fact that the inputs sent are keyboard/gamepad based (sending direct inputs from the keyboard and controller) and not button based (only sending the buttons configured in the Input Settings).~~
(Only an issue in https://github.com/K4thos/Ikemen_GO)

Thank you for your time.

**Comment by theMadness** [comment]
Step 1 here would be porting GGPO to golang, I assume.
I might look into that myself.

Please let me know if this assumption is unfounded.

**Comment by Kikasuru** [comment]
> Step 1 here would be porting GGPO to golang, I assume.
> I might look into that myself.
> 
> Please let me know if this assumption is unfounded.

[Golang has a couple ways of using a library made in C++.](https://stackoverflow.com/questions/1713214/how-to-use-c-in-go)

**Comment by theMadness** [comment]
It does look like there's a few drawbacks tho.

**Comment by Kikasuru** [comment]
Another statement on this, rollback is fairly simple to understand imo, you wouldn't even need to use the GGPO library if you don't want to.

[Core-A Gaming does an amazing explanation of it.](https://www.youtube.com/watch?v=0NLe4IpdS1w)

**Comment by Windblade-GR01** [comment, marked as answer]
So a contributor is implementing rollback.
No estimated date of release but someone is working on it.

**Comment by FerLuisxd** [comment]
Any updates on this?
